addons/events/src/components/Event.js

```diff
@@ -133,14 +134,14 @@ class Item extends Component {
   };
 
   static getDerivedStateFromProps = ({ payload }, { prevPayload }) => {
-    if (payload !== prevPayload) {
+    if (!isEqual(payload, prevPayload)) {
       const payloadString = json.plain(payload);
-
+      const refinedPayload = getJSONFromString(payloadString);
       return {
         failed: false,
-        payload: getJSONFromString(payloadString),
+        payload: refinedPayload,
         payloadString,
-        prevPayload,
+        prevPayload: refinedPayload,
```

Comment:
Why are payload & prevPayload the same thing?

Reply:
The reason for why `payload` and `prevPayload` exist is to determine whether the payload is changed or not(which is a prop and coming from user input from story)

This assigning value to `prevPayload` is only executed when coming `payload` and previous `payload`(which is the same as `prevPayload`)

Reply:
@shilman I'm wondering if my explanation is not clear? just double-check